src/actions/deriveAddresses.ts

```diff
@@ -134,6 +135,25 @@ function getNativeSegWitAddress(
   return String(address);
 }
 
+/**
+ * derive a Taproot address at a given account and index positions
+ * @param xpub the xpub from which to derive the native SegWit address
+ * @param account account number from which to derive the native SegWit address
+ * @param index index number from which to derive the native SegWit address
+ * @returns the derived Taproot address
+ */
+ function getTaprootAddress(
+  xpub: string,
+  account: number,
+  index: number,
+): string {
+
+  var account1 = new BIP86.fromXPub(xpub);
+  var address = account1.getAddress(index,!!account);
```

Comment:
Why would you do `!!acount`?
It coalesces the value to either a 0 or a 1, I don't think that's what you wanted to do?

Reply:
account has 0 or 1
!!account makes it false or true - this is expected input for getAddress function of BIP86 package which I used for getting taproot addressed.
for legacy, segwit adresses other package was used which required 0 or 1 as input
so this was by intention.
I think getting segwit and legacy addresses should be refactored btw to a more simple form, like here.